RESOURCE-121 Support azure service bus topics (#565)
inspec/inspec-azure: docs/resources/azure_service_bus_topics.md

---
title: About the azure_service_bus_topics Resource
platform: azure
---

# azure_service_bus_topics

Use the `azure_service_bus_topics` InSpec audit resource to test properties related to all Azure Service Bus Topics within a project.

## Azure REST API version, endpoint and http client parameters

This resource interacts with api versions supported by the resource provider.
The `api_version` can be defined as a resource parameter.
If not provided, the latest version will be used.
For more information, refer to [`azure_generic_resource`](azure_generic_resource.md).

Unless defined, `azure_cloud` global endpoint, and default values for the http client will be used.
For more information, refer to the resource pack [README](../../README.md).

## Availability

### Installation

This resource is available in the [InSpec Azure resource pack](https://github.com/inspec/inspec-azure).
For an example `inspec.yml` file and how to set up your Azure credentials, refer to resource pack [README](../../README.md#Service-Principal).

## Syntax

An `azure_service_bus_topics` resource block returns all Azure Service Bus Topics within a project.

> `name`, `namespace_name` and `resource_group` is a required parameter.

Reviewer: ```suggestion
`name`, `namespace_name`, and `resource_group` are the required parameters.
```
